sdk/src/Core/Amazon.Util/AWSSDKUtils.cs

```diff
@@ -128,7 +129,7 @@ private static string DetermineValidPathCharacters()
         {
             const string basePathCharacters = "/:'()!*[]$";
 
-            var sb = new StringBuilder();
+            var sb = new ValueStringBuilder(basePathCharacters.Length * 2);
```

Comment:
For each of these usages it might make sense to help me to find reasonable defaults.